pkg/routeagent_driver/handlers/ovn/avoid_snat_handler.go

```diff
@@ -0,0 +1,251 @@
+	subnets := extractIPv4Subnets(&endpoint.Spec)
+	for _, subnet := range subnets {
+		err := h.localIPSet.AddEntry(subnet, true)
+		if err != nil {
+			return errors.Wrap(err, "error adding local IP set entry")
+		}
+	}
+
+	return nil
+}
+
+func (h *AvoidSNATHandler) LocalEndpointRemoved(endpoint *submV1.Endpoint) error {
+	subnets := extractIPv4Subnets(&endpoint.Spec)
+	for _, subnet := range subnets {
+		logError(h.localIPSet.DelEntry(subnet), "Error deleting the subnet %q from the local IPSet", subnet)
+	}
+
+	return nil
+}
+
+func (h *AvoidSNATHandler) RemoteEndpointCreated(endpoint *submV1.Endpoint) error {
+	subnets := extractIPv4Subnets(&endpoint.Spec)
+	for _, subnet := range subnets {
+		err := h.remoteIPSet.AddEntry(subnet, true)
+		if err != nil {
+			return errors.Wrap(err, "error adding remote IP set entry")
+		}
+	}
+
+	return nil
+}
+
+func (h *AvoidSNATHandler) RemoteEndpointRemoved(endpoint *submV1.Endpoint) error {
+	subnets := extractIPv4Subnets(&endpoint.Spec)
+	for _, subnet := range subnets {
+		logError(h.remoteIPSet.DelEntry(subnet), "Error deleting the subnet %q from the remote IPSet", subnet)
+	}
+
+	return nil
+}
+
+func (h *AvoidSNATHandler) Uninstall() error {
+	logger.Infof("Flushing packetfilter entry in %q chain of %q table", constants.SmSelfSnatChain, constants.NATTable)
+
+	if err := h.pFilter.ClearChain(packetfilter.TableTypeNAT, constants.SmSelfSnatChain); err != nil {
+		logger.Errorf(err, "Error flushing packetfilter chain %q of %q table", constants.SmSelfSnatChain,
+			constants.NATTable)
+	}
+
+	logger.Infof("Deleting packetfilter entry in %q chain of %q table", constants.SmSelfSnatChain, constants.NATTable)
+
+	logError(h.pFilter.DeleteIPHookChain(&packetfilter.ChainIPHook{
+		Name:     constants.SmSelfSnatChain,
+		Type:     packetfilter.ChainTypeNAT,
+		Hook:     packetfilter.ChainHookPostrouting,
+		Priority: packetfilter.ChainPriorityMiddle,
+	}), "Error deleting IP hook chain %q of table type %q", constants.SmSelfSnatChain, packetfilter.ChainTypeNAT)
+
+	logger.Infof("Flushing packetfilter entries in %q chain of %q table", constants.SmSelfSnatChain, constants.NATTable)
+
+	logError(h.localIPSet.Flush(), "Error flushing ipset %q", h.localCIDRIPSet)
+
+	logError(h.localIPSet.Destroy(), "Error deleting ipset %q", h.localCIDRIPSet)
+
+	logError(h.remoteIPSet.Flush(), "Error flushing ipset %q", h.remoteCIDRIPSet)
+
+	logError(h.remoteIPSet.Destroy(), "Error deleting ipset %q", h.remoteCIDRIPSet)
+
+	return nil
+}
+
+func logError(err error, format string, args ...interface{}) {
+	if err != nil {
+		logger.Errorf(err, format, args...)
+	}
+}
+
+func extractIPv4Subnets(endpoint *submV1.EndpointSpec) []string {
+	subnets := make([]string, 0, len(endpoint.Subnets))
+
+	for _, subnet := range endpoint.Subnets {
+		// Revisit when IPv6 support is added.
+		if k8snet.IsIPv4CIDRString(subnet) {
+			subnets = append(subnets, subnet)
+		}
+	}
+
+	return subnets
```

Comment:
```suggestion
	return cidr.ExtractIPv4Subnets(endpoint.Subnets)
```

This could also just be inlined at the call sites.